src/classes/SRVC_Contact_PreventDeletion.cls

```diff
@@ -0,0 +1,121 @@
+            if (contactHasChildRecords(eachContact)) {
+                Contact contactInContext = contactById.get(eachContact.ID);
+                contactInContext.addError(Label.CannotDelete);
+                contactPreventedFromDeletionList.add(contactInContext);
+            }
+        }
+
+        return contactPreventedFromDeletionList;
+    }
+
+    /****************************************************************************************************************
+    @description Returns a list of Contacts and assocaited child records
+    @params contactIdSet A set of Contact Id to return the records for
+    @return List<Contact>
+    *****************************************************************************************************************/
+    @TestVisible
+    private List<Contact> getContactsAndChildRelationships(set<Id> contactIdSet) {
+        return [SELECT ID,
+                (SELECT ID FROM Addresses__r LIMIT 1),
+                (SELECT ID FROM Affl_Accounts__r LIMIT 1),
+                (SELECT ID FROM Applications_Applicant__r LIMIT 1),
+                (SELECT ID FROM Applications_Preparer__r LIMIT 1),
+                (SELECT ID FROM Attendance_Events__r LIMIT 1),
+                (SELECT ID FROM Attributes__r LIMIT 1),
+                (SELECT ID FROM Behavior_Involvements__r LIMIT 1),
+                (SELECT ID FROM Contact_Languages__r LIMIT 1),
+                (SELECT ID FROM Courses_Taught__r LIMIT 1),
+                (SELECT ID FROM Program_Enrollments__r LIMIT 1),
+                (SELECT ID FROM Student_Course_Enrollments__r LIMIT 1),
+                (SELECT ID FROM Term_Grades__r LIMIT 1),
+                (SELECT ID FROM Tests__r LIMIT 1)
+                FROM Contact
+                WHERE ID IN :contactIdSet];
+    }
+
+   /*******************************************************************************************************
+   * @description Evaluates whether the Contact has any child related records.
+   * @param con is the current Contact record.
+   * @return Boolean.
+   ********************************************************************************************************/
+    @TestVisible
+    protected virtual Boolean contactHasChildRecords(Contact con) {
+        return (con.Addresses__r.isEmpty() == false || con.Affl_Accounts__r.isEmpty() == false ||
+                con.Applications_Applicant__r.isEmpty() == false  || con.Applications_Preparer__r.isEmpty() == false ||
+                con.Attendance_Events__r.isEmpty() == false || con.Attributes__r.isEmpty() == false ||
+                con.Behavior_Involvements__r.isEmpty() == false || con.Contact_Languages__r.isEmpty() == false ||
+                con.Courses_Taught__r.isEmpty() == false || con.Program_Enrollments__r.isEmpty() == false ||
+                con.Student_Course_Enrollments__r.isEmpty() == false || con.Term_Grades__r.isEmpty() == false ||
+                con.Tests__r.isEmpty() == false);
```

Comment:
Given the size of this, I feel it is easier to scan vertically than the combination of horizontally and vertically. This is not a requirement, simply a suggestion. Perhaps it is born out of personal preference.
```suggestion
        return (con.Addresses__r.isEmpty() == false ||
                con.Affl_Accounts__r.isEmpty() == false ||
                con.Applications_Applicant__r.isEmpty() == false ||
                con.Applications_Preparer__r.isEmpty() == false ||
                con.Attendance_Events__r.isEmpty() == false ||
                con.Attributes__r.isEmpty() == false ||
                con.Behavior_Involvements__r.isEmpty() == false ||
                con.Contact_Languages__r.isEmpty() == false ||
                con.Courses_Taught__r.isEmpty() == false ||
                con.Program_Enrollments__r.isEmpty() == false ||
                con.Student_Course_Enrollments__r.isEmpty() == false ||
                con.Term_Grades__r.isEmpty() == false ||
                con.Tests__r.isEmpty() == false);
```